encoding/thrift/thriftrw-plugin-yarpc/server.go

```diff
@@ -149,6 +163,8 @@ func (h handler) <.Name>(ctx <$context>.Context, body <$wire>.Value) (<$thrift>.
 	if err == nil {
 		response.IsApplicationError = hadError
 		response.Body = result
+	} else {
```

Comment:
This logic isn't quite right. Setting the error name belongs within the same `err == nil` condition.
Here, the checked error is not from the handler, it's actually from the `WrapResponse` [which looks like this](https://sourcegraph.com/github.com/yarpc/yarpc-go@0110ecb6e3ef9f86f152acd0badbd4bd5c87166b/-/blob/encoding/thrift/thriftrw-plugin-yarpc/internal/tests/atomic/atomic.go#L1274).


```suggestion
```